src/pudl/output/sec10k.py

```diff
@@ -21,6 +22,111 @@ def _filename_sec10k_to_source_url(
     return "https://www.sec.gov/Archives/edgar/data/" + filename_sec10k + ".txt"
 
 
+def _fill_sics(df: pd.DataFrame) -> pd.DataFrame:
+    """Fill missing SIC IDs and names where possible.
+
+    Within the reporting history for each company, as identified by Central Index Key
+    (CIK), fill in missing values of ``industry_id_sic`` when the values reported before
+    and after the gap are the same. If the beginning of the series is missing, backfill
+    it. If the end of the series is missing, forward fill it. Assign industry groups and
+    specific industry names based on the canonical descriptions from the SEC which
+    correspond to the reported industry ID.
+
+    This step is deferred until the output table because it requires the ``report_date``
+    column, which is not part of the ``core_sec10k__quarterly_company_information``
+    table
+    """
+    sorted_sics = df.sort_values(by=["central_index_key", "report_date"]).assign(
+        # 1044 is "silver mines" which doesn't show up in the canonical SEC list.
+        # Instead it should be 1040 "gold and silver mines" -- there's only one
+        # company that used the overly specific SIC.
+        industry_id_sic=lambda _df: _df["industry_id_sic"].replace({"1044": "1040"})
+    )
+    non_null_sics = sorted_sics.loc[
+        :, ["industry_id_sic", "industry_name_sic"]
+    ].dropna()
+    assert not non_null_sics.empty, "No non-null SICs to validate!"
+
+    # Check that the reported mapping between SIC IDs and names is unique -- that each
+    # SIC ID maps to at most one SIC name.
+    sic_nunique_names = (
+        non_null_sics.groupby("industry_id_sic")["industry_name_sic"]
+        .nunique()
+        .reset_index()
+    )
+    # 2820 has one entry with garbled spelling.
+    # 1040 we just munged up above.
+    # The industry names associated with them will be overwritten by the official
+    # SEC list.
+    known_nonunique_sics = {"1040", "2820"}
+    non_unique_sics = sic_nunique_names[sic_nunique_names["industry_name_sic"] > 1][
+        "industry_id_sic"
+    ].to_list()
+    assert set(non_unique_sics) == known_nonunique_sics, (
+        f"Found non-unique mapping of SIC ID to industry name! {non_unique_sics}"
```

Comment:
If this assert fails, the first thing we'll need to do is figure out which non-unique SICs were found that were not expected, so might as well do that in the error message itself

```suggestion
        f"Found non-unique mapping of SIC ID to industry name! {non_unique_sics-known_nonunique_sics}"
```